### ISSUE TITLE
DataCloneError on outExtension when DTS is on

### ISSUE THREAD

**Issue by zvictor** [egoist/tsup#667]
While using the config file shown below, I get a `DataCloneError` whenever I turn `dts` on.

```ts
import { defineConfig } from 'tsup'

export default defineConfig({
  format: ['esm', 'cjs'],
  // dts: true,
  outExtension: ({ format }) => ({
    js: `.${format === 'esm' ? 'mjs' : format}`,
  }),
})
```

Error when `dts` is on:
```haskell
DataCloneError: ({ format }) => ({
    js: `.${format === "esm" ? "mjs" : format}`
  }) could not be cloned.
    at new DOMException (node:internal/per_context/domexception:53:5)
    at Worker.postMessage (node:internal/worker:342:5)
    at ~/faugra/node_modules/.pnpm/tsup@6.1.2_typescript@4.7.4/node_modules/tsup/dist/index.js:2060:18
    at new Promise (<anonymous>)
    at dtsTask (~/faugra/node_modules/.pnpm/tsup@6.1.2_typescript@4.7.4/node_modules/tsup/dist/index.js:2058:15)
    at ~/faugra/node_modules/.pnpm/tsup@6.1.2_typescript@4.7.4/node_modules/tsup/dist/index.js:2171:24
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0)
    at async build (~/faugra/node_modules/.pnpm/tsup@6.1.2_typescript@4.7.4/node_modules/tsup/dist/index.js:2046:3)
    at async CAC.<anonymous> (~/faugra/node_modules/.pnpm/tsup@6.1.2_typescript@4.7.4/node_modules/tsup/dist/chunk-HL6TILEC.js:185:5)
```

**Comment by zvictor** [comment]
Another thing disturbing me is the fact that after this error is thrown the process keeps running forever.
I don't get an exit code, so my [`toThrow` tests](https://jestjs.io/docs/expect#tothrowerror) don't catch it at all.

**Comment by egoist** [comment]
fixed in 6.1.3
